fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergRESTCatalog.java

```diff
@@ -0,0 +1,75 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.connector.iceberg;
+
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.IcebergTable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.rest.RESTCatalog;
+import org.apache.thrift.TException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import static com.starrocks.connector.iceberg.IcebergUtil.convertToSRDatabase;
+
+public class IcebergRESTCatalog extends RESTCatalog implements IcebergCatalog {
+
+    private static final ConcurrentHashMap<String, IcebergRESTCatalog> REST_URI_TO_CATALOG =
+            new ConcurrentHashMap<>();
+
+    public static synchronized IcebergRESTCatalog getInstance(Map<String, String> properties) {
+        String uri = properties.get(CatalogProperties.URI);
+        if (!REST_URI_TO_CATALOG.containsKey(uri)) {
```

Comment:
use `computeIfAbsent` instead.

Reply:
Yeah I agree, I was keeping it consistent w/ the other catalogs but I'll update it here

Reply:
This is done.